### ISSUE TITLE
Dashboard variables disappear on time range change (Loki)

### ISSUE THREAD

**Issue by maksimse** [grafana/grafana#93469]
### What happened?

Dashboard variables disappear after time range change.
Variables are loki-based labels. They are set up to refresh on time range change.
In order to get them back, I need to refresh the page or hit the refresh button on right-hand side.

Video:
https://github.com/user-attachments/assets/5af2ab74-0f13-4f26-a52a-a85ab982ec7f

### What did you expect to happen?

See present variable values.

### Did this work before?

It worked before on Grafana v8.5.9.
We are now running v11.2.0 and Loki v3.1.1. It is the same with Loki v2.8.7 .

### How do we reproduce it?

1. Install Loki v3.1.1
2. Install Grafana v11.2.0
3. Populate loki with logs with the same labels as in the video.
4. Create a similar dashboard to view logs.
5. Try to change time range between 5m, 15m, 1h.

### Is the bug inside a dashboard panel?

| Key | Value |
|--|--|
| Panel | logs @ Grafana v11.2.0 (unknown-de) |
| Grafana | Grafana v11.2.0 (unknown-de) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "description": "",
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "showTime": true,
        "showLabels": false,
        "showCommonLabels": true,
        "wrapLogMessage": true,
        "prettifyLogMessage": false,
        "enableLogDetails": true,
        "dedupStrategy": "none",
        "sortOrder": "Descending"
      },
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "typeVersion": [
                    0,
                    0
                  ],
                  "custom": {
                    "frameType": "LabeledTimeValues",
                    "lokiQueryStatKey": "Summary: total bytes processed"
                  },
                  "stats": [
                    {
                      "displayName": "Summary: bytes processed per second",
                      "unit": "Bps",
                      "value": 10369
                    },
                    {
                      "displayName": "Summary: lines processed per second",
                      "value": 288
                    },
                    {
                      "displayName": "Summary: total bytes processed",
                      "unit": "decbytes",
                      "value": 72
                    },
                    {
                      "displayName": "Summary: total lines processed",
                      "value": 2
                    },
                    {
                      "displayName": "Summary: exec time",
                      "unit": "s",
                      "value": 0.006944
                    },
                    {
                      "displayName": "Ingester: total reached",
                      "value": 2
                    },
                    {
                      "displayName": "Ingester: total chunks matched",
                      "value": 1
                    },
                    {
                      "displayName": "Ingester: total batches",
                      "value": 3
                    },
                    {
                      "displayName": "Ingester: total lines sent",
                      "value": 2
                    },
                    {
                      "displayName": "Ingester: head chunk bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: head chunk lines",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: decompressed bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: decompressed lines",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: compressed bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: total duplicates",
                      "value": 0
                    }
                  ],
                  "executedQueryString": "Expr: {subsystem=\"esrp\", zone=\"us-east-1a\", unit=\"\", hostname=\"esrp-1a.stage.nga911.com\", priority=\"5\"} |= \"\"",
                  "preferredVisualisationType": "logs",
                  "limit": 5000,
                  "searchWords": [
                    "\\$search"
                  ]
                },
                "fields": [
                  {
                    "name": "labels",
                    "type": "other",
                    "typeInfo": {
                      "frame": "json.RawMessage"
                    },
                    "config": {}
                  },
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {}
                  },
                  {
                    "name": "Line",
                    "type": "string",
                    "typeInfo": {
                      "frame": "string"
                    },
                    "config": {}
                  },
                  {
                    "name": "tsNs",
                    "type": "string",
                    "typeInfo": {
                      "frame": "string"
                    },
                    "config": {}
                  },
                  {
                    "name": "labelTypes",
                    "type": "other",
                    "typeInfo": {
                      "frame": "json.RawMessage"
                    },
                    "config": {
                      "custom": {
                        "hidden": true
                      }
                    }
                  },
                  {
                    "name": "id",
                    "type": "string",
                    "typeInfo": {
                      "frame": "string"
                    },
                    "config": {}
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    {
                      "app": "db",
                      "client": "stage",
                      "hostname": "esrp-1a.stage.nga911.com",
                      "instanceid": "i-03a639bb4e59f53bc",
                      "job": "systemd-journal",
                      "priority": "5",
                      "region": "us-east-1",
                      "service_name": "db",
                      "subsystem": "esrp",
                      "zone": "us-east-1a"
                    },
                    {
                      "app": "db",
                      "client": "stage",
                      "hostname": "esrp-1a.stage.nga911.com",
                      "instanceid": "i-03a639bb4e59f53bc",
                      "job": "systemd-journal",
                      "priority": "5",
                      "region": "us-east-1",
                      "service_name": "db",
                      "subsystem": "esrp",
                      "zone": "us-east-1a"
                    }
                  ],
                  [
                    1726736461527,
                    1726736461520
                  ],
                  [
                    "(/etc/cron.hourly) finished 0anacron",
                    "(/etc/cron.hourly) starting 0anacron"
                  ],
                  [
                    "1726736461527304000",
                    "1726736461520959000"
                  ],
                  [
                    {
                      "app": "I",
                      "client": "I",
                      "hostname": "I",
                      "instanceid": "I",
                      "job": "I",
                      "priority": "I",
                      "region": "I",
                      "service_name": "I",
                      "subsystem": "I",
                      "zone": "I"
                    },
                    {
                      "app": "I",
                      "client": "I",
                      "hostname": "I",
                      "instanceid": "I",
                      "job": "I",
                      "priority": "I",
                      "region": "I",
                      "service_name": "I",
                      "subsystem": "I",
                      "zone": "I"
                    }
                  ],
                  [
                    "1726736461527304000_31f890ec",
                    "1726736461520959000_d83ae236"
                  ]
                ],
                "nanos": [
                  null,
                  [
                    304000,
                    959000
                  ],
                  null,
                  null,
                  null,
                  null
                ]
              }
            }
          ]
        }
      ],
      "title": "Reproduced with embedded data",
      "type": "logs"
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >logs @ Grafana v11.2.0 (unknown-de)</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[loki]</td>\n    </tr>\n    \n    <tr><th>Data</th><td> 1 frames, 6 fields, 2 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>Grafana v11.2.0 (unknown-de) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"datasource\": {\n    \"type\": \"loki\",\n    \"uid\": \"${datasource}\"\n  },\n  \"description\": \"\",\n  \"gridPos\": {\n    \"h\": 27,\n    \"w\": 24,\n    \"x\": 0,\n    \"y\": 0\n  },\n  \"id\": 2,\n  \"options\": {\n    \"showTime\": true,\n    \"showLabels\": false,\n    \"showCommonLabels\": true,\n    \"wrapLogMessage\": true,\n    \"prettifyLogMessage\": false,\n    \"enableLogDetails\": true,\n    \"dedupStrategy\": \"none\",\n    \"sortOrder\": \"Descending\"\n  },\n  \"targets\": [\n    {\n      \"datasource\": {\n        \"type\": \"loki\",\n        \"uid\": \"${datasource}\"\n      },\n      \"expr\": \"{subsystem=\\\"$subsystem\\\", zone=\\\"$zone\\\", unit=\\\"$unit\\\", hostname=\\\"$hostname\\\", priority=\\\"$priority\\\"} |= \\\"$search\\\"\",\n      \"maxLines\": 5000,\n      \"queryType\": \"range\",\n      \"refId\": \"A\"\n    }\n  ],\n  \"title\": \"Logs for $unit\",\n  \"type\": \"logs\"\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: Logs for $unit // 2024-09-19 12:18:11",
  "tags": [
    "debug",
    "debug-logs"
  ],
  "time": {
    "from": "2024-09-19T08:18:11.344Z",
    "to": "2024-09-19T09:18:11.344Z"
  }
}
```
</details>

### Environment (with versions)?

Grafana: `11.2.0`
OS: `Rocky Linux 9.4 (Blue Onyx) 5.14.0-427.35.1.el9_4.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Sep 12 18:24:53 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
Browser: `Google Chrome 128.0.6613.113 (arm64)`

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

Loki 5.0.0

**Comment by maksimse** [comment]
I downgraded to v11.1.5 today and the issue still remains.
Since we are compiling Grafana from sources, I tried also https://dl.grafana.com/oss/release/grafana-11.2.0-1.x86_64.rpm to exclude any issues during compilation.
It didn't help - the issue is still there.

**Comment by maksimse** [comment]
![Image](https://github.com/user-attachments/assets/ce3d45f5-56fb-46fc-b362-4044e1f6fdba)
I get this error in Google chrome dev tools. Is this the issue?
I don't get this error in Grafana 8.5.9 .

**Comment by maksimse** [comment]
This issue has been resolved in #93931 .
I have just compiled Grafana 11.3.0-pre and can confirm that the issue is gone. Filter is working properly.
